var/spack/repos/builtin/packages/sherpa/package.py

```diff
@@ -261,3 +263,17 @@ def flag_handler(self, name, flags):
                     flags.append("-m64")
 
         return (None, None, flags)
+
+    # This may not be needed when this package is changed to be a CMake package
+    # since it's specific to makelibs
+    def install(self, spec, prefix):
+        # Remove the path to the spack compiler wrappers
+        # We don't want to use the paths to the wrappers later on
+        filter_file(r".*/spack/lib/spack.*", "", "AMEGIC++/Main/makelibs")
```

Comment:
The first `spack` in the path is where you may have installed spack but it is not a guarantee. I have (work)trees with e.g. `spack-v0.21.2` instead. It should be SPACK_ROOT.

Reply:
In fact, you may want to use the spack function for this, e.g. https://github.com/spack/spack/blob/07fb83b493d522ef5369e4db565841674b4512c1/var/spack/repos/builtin/packages/rivet/package.py#L73

Reply:
I didn't know about `filter_compiler_wrappers`; I changed the first `filter_file` to that but I haven't tested it.